### ISSUE TITLE
[SPIR-V] Rasterizer order view is not supported

### ISSUE THREAD

**Issue by LiamTyler** [microsoft/DirectXShaderCompiler#5124]
Hello,

I have a HLSL shader from Unreal Engine 5. It compiles fine to DXIL when using DXC, but if I add the `-spirv` argument, it crashes with this message.

`Internal compiler error: access violation. Attempted to write from address 0x0000000000000060`

Attached [bad_shader.zip](https://github.com/microsoft/DirectXShaderCompiler/files/11116838/bad_shader.zip) is the shader, and the command line args im using are: `-spirv -T ps_6_6 -E MainPS bad_shader.hlsl`

**Comment by Keenuts** [comment]
Hi, thanks for the report!
Was able to reproduce the error.
Seems like there are 2 issues:
 - we deref a pointer without checking
 - this code hits a non-implemented path.
 
 Sending a PR to fix the crash, don't know yet if the fast that we reach an non-implemented path is still normal
 
 Compiler error after fixing the crash:
 ```
 /Engine/Private/BasePassPixelShader.usf:752:48: warning: ambiguous type for bit shift; use a type suffix on literal values, like 'L' or 'U', or a cast [-Wambig-lit-shift]
                        OcclusionMask |= bIsBentNormalOccluded ? (1 << TracingRayIndex) : 0;
                                                                    ^
bad_shader.hlsl:3938:35: error: initializer for type 'RasterizerOrderedTexture2D<unsigned int>' unimplemented
        RasterizerOrderedTexture2D<uint> OutOITSampleCount;
 ```

**Comment by Keenuts** [comment]
Minimal repro:

```hlsl
RasterizerOrderedTexture2D<uint> rot;

static const struct {
	RasterizerOrderedTexture2D<uint> rot_field;
} cstruct = {rot};


void MainPS()
{ }
```
dxc -T ps_6_6 -E MainPS -spirv -Od -Vd repro.hlsl
```

```

**Comment by s-perron** [comment]
This is a bigger issue to fix. The SPIR-V path does not support [rasterizer order views](https://learn.microsoft.com/en-us/windows/win32/direct3d11/rasterizer-order-views). A proper fix will be to implement them all, and I do not have a sizing for it.

**Comment by LiamTyler** [comment]
Okay. So to clarify, it sounds like the fix is a long ways out? I just pulled this shader randomly from UE5, im sure there are plenty more similar to it. It would be nice to be able to get SPIRV out of all of those shaders

**Comment by s-perron** [comment]
This might be a bigger piece of work that we do not have planned. I think this is a ways out.

Notes for whoever implements [rasterizer order views](https://learn.microsoft.com/en-us/windows/win32/direct3d11/rasterizer-order-views). We will need to use this Vulkan extension. 
https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_EXT_fragment_shader_interlock.html.

We will have to create a critical section around the operations of the operations on these type. I believe we will want to use the `FragmentShaderPixelInterlockEXT` for the order.

**Comment by cassiebeckley** [comment]
Fixed by #5820.
